codegenerator/cli/templates/static/codegen/src/EventProcessing.res

```diff
@@ -133,7 +133,7 @@ let runEventContractRegister = (
       )
 
     let val = switch (dynamicContracts, dynamicContractRegistrations) {
-    | ([], dynamicContractRegistrations) => dynamicContractRegistrations
+    | ([], None) => None
```

Comment:
To be clear, it was hitting this case when the `contractRegister` function produced no registrations (possibly because of duplicate registration)

AND the current `dynamicContratRegistrations` was `Some`. Then it would just pass through the current registrations without adding itself to the unprocessed batch. Which we only want on the None case.